src/common/shared_lib.h

```diff
@@ -0,0 +1,43 @@
+// This provides a portable way of consuming shared libraries,
+// staying as close as possible to dlopen semantics.
+
+#ifndef SHARED_LIB_H_
+#define SHARED_LIB_H_
+
+#if !defined(_WIN32)
+  #include <dlfcn.h>
+#endif
+
+#ifdef __APPLE__
+#define SHARED_LIB_SUFFIX ".dylib"
+#elif defined(_WIN32)
+#define SHARED_LIB_SUFFIX ".dll"
+#else
+#define SHARED_LIB_SUFFIX ".so"
+#endif
+
+#ifdef _WIN32
+  #include <windows.h>
+  typedef HMODULE lib_handle;
+#else
+  #include <ostream>
+  typedef void* lib_handle;
```

Comment:
Typedefs are usually looking like `some_type_t`
And in this case I'd call it `shared_lib_handle_t`

Reply:
``shared_lib_handle_t`` sounds good.